draft-aegis-mimi-arch-latest.md

```diff
@@ -540,14 +543,15 @@ A gateway MUST allow an application or gateway to retrieve a key package of a
 given client using the following request:
 
 ~~~
-HTTP POST /apps/{appId}/clients/{clientId}/keyPackages
+HTTP GET /apps/{appId}/clients/{clientId}/keyPackage
 
 Input:
 
 {
     protocolVersion: uint16,
     cipherSuite: uint16,
     optional appLabel: String
+    receiver_client_id: String
```

Comment:
Are we trusting the fetching client/application  back end to supply the correct receiver_client_id? Is there anything stoping Alice from fetching the KP assigned to Charlie if she knows Charlie's clientId?